examples/gauges/README.md

```diff
@@ -0,0 +1,289 @@
+# Gauges Example
+
+
+Table of Contents
+=================
+
+* [Gauges](#gauges)
+    * [Run the example](#run-the-example)
+    * [How to use gauges?](#how-to-use-gauge?)
+        * [Initialize metric registry](#initialize-metric-registry)
+        * [Create gauge metric](#create-gauge-metric)
+        * [Create gauge entry](#create-gauge-entry)
+        * [Set gauge value](#set-gauge-values)
+        * [Complete example](#complete-example)
+
+# Summary 
+[top](#Table-of-Contents)
+
+This example shows how to use gauge metrics. The program records two gauges.
+
+1. **process_heap_alloc (int64)**: Total bytes used by objects allocated in the heap. It includes objects currently used and objects that are freed but not garbage collected.
+1. **process_heap_idle_to_alloc_ratio (float64)**: It is the ratio of Idle bytes to allocated bytes in the heap.
+
+It periodically runs a function that retrieves the memory stats and updates the above two metrics. These metrics are then exported using prometheus exporter.
+Metrics can be viewed at [http://localhost:9090/metrcs](http://localhost:9090/metrcs) once the program is running.
+The program lets you choose the amount of memory (in MB) to consume. Choose different values and query the metrics to see the change in metrics.
+
+## Run the example.
+
+```
+$ go get go.opencensus.io/examples/gauges/...
+```
+
+then:
+
+```
+$ go run $(go env GOPATH)/src/go.opencensus.io/examples/gauges/gauge.go
+```
+
+## How to use gauge?
+
+### Initialize Metric Registry
+Create a new metric registry for all your metrics.
+This step is a general step for any kind of metrics and not specific to gauges.
+Register newly created registry with global producer manager.
+
+[embedmd]:# (gauge.go reg)
+```go
+r := metric.NewRegistry()
+metricproducer.GlobalManager().AddProducer(r)
+```
+
+
+### Create gauge metric
+Create a gauge metric. In this example we have two metrics.
+
+**process_heap_alloc**
+
+[embedmd]:# (gauge.go alloc)
+```go
+allocGauge, err := r.AddInt64Gauge(
+	"process/heap/alloc",
```

Comment:
Nit: Prometheus doesn't allow '/', consider replacing with '_' instead.

Reply:
prometheus exporter replaces '/' with '_'.   
Should we '/' or '_' be based on exporter? or based on what is more appropriate generically?

Reply:
In this example I'm inclined to be consistent with the exporter, since it will be confusing if users see the '/' characters being swapped on Prometheus endpoint. In addition the readme suggests the metric names use '_' ("process_heap_alloc"): https://github.com/census-instrumentation/opencensus-go/pull/1107/files#diff-2b3521dd2010161fca962a345a4eaddeR21.

Reply:
SG